lib/auth/dbobject/dbobjectv1/service.go

```diff
@@ -0,0 +1,224 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package dbobjectv1
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/gravitational/teleport"
+	pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/dbobject/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/authz"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/srv/db/common/databaseobject"
+)
+
+// Backend interface for manipulating DatabaseObject resources.
+type Backend interface {
+	services.DatabaseObjects
+}
+
+// DatabaseObjectServiceConfig holds configuration options for
+// the DatabaseObjects gRPC service.
+type DatabaseObjectServiceConfig struct {
+	Authorizer authz.Authorizer
+	Backend    Backend
+	Logger     logrus.FieldLogger
+}
+
+// NewDatabaseObjectService returns a new instance of the DatabaseObjectService.
+func NewDatabaseObjectService(cfg DatabaseObjectServiceConfig) (*DatabaseObjectService, error) {
+	if cfg.Authorizer == nil {
+		return nil, trace.BadParameter("authorizer is required")
+	}
+	if cfg.Backend == nil {
+		return nil, trace.BadParameter("backend service is required")
+	}
+	if cfg.Logger == nil {
+		cfg.Logger = logrus.WithField(teleport.ComponentKey, "db_object")
+	}
+	return &DatabaseObjectService{
+		logger:     cfg.Logger,
+		authorizer: cfg.Authorizer,
+		backend:    cfg.Backend,
+	}, nil
+}
+
+// DatabaseObjectService implements the teleport.dbobject.v1.DatabaseObjectService RPC service.
+type DatabaseObjectService struct {
+	// UnsafeDatabaseObjectServiceServer is embedded to opt out of forward compatibility for this service.
+	// Added methods to DatabaseObjectServiceServer will result in compilation errors, which is what we want.
+	pb.UnsafeDatabaseObjectServiceServer
+
+	backend    Backend
```

Comment:
Do we want to use the cache for some calls ? Or are we happy that this will be sufficiently low traffic ?

Reply:
I want to look into cache in the future PR, this one is already pretty large.

Also, right now there are no readers. This will change in the near future, so I'll make sure to address it one way or another.

Reply:
Sounds good to me.